azurerm/internal/services/storage/storage_account_resource_test.go

```diff
@@ -1954,3 +2037,164 @@ resource "azurerm_storage_account" "test" {
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomString)
 }
+
+func (r StorageAccountResource) azureFilesIdentityBasedAuthenticationAADDS(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-storage-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "unlikely23exst2acct%s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+
+  azure_files_identity_based_authentication {
+    directory_service_options = "AADDS"
```

Comment:
When testing this, I get the following 409 error:
```json
{
  "error": {
    "code": "AadDsTenantNotFound",
    "message": "Unable to locate active AAD DS for AAD tenant Id 26e25406-6564-4a26-98ee-c71ba03235ad associated with the storage account. For more information, see - https://aka.ms/storagefilesaad"
  }
}
```

Presumably this means AADDS must be set up first? If that's the case, that would unfortunately make this untestable since there's an outstanding API issue https://github.com/Azure/azure-rest-api-specs/issues/11200 preventing us from configuring AADDS.